File: dart/difference-of-squares/lib/difference_of_squares.dart
import 'dart:math';

class DifferenceOfSquares {
  num squareOfSum(int input) => pow(((input * (input + 1)) ~/ 2), 2);
  num sumOfSquares(int input) => ((input * (input + 1) * (2 * input + 1)) ~/ 6);
  num differenceOfSquares(int input) =>
      squareOfSum(input) - sumOfSquares(input);
}

File: typescript/difference-of-squares/difference-of-squares.ts
export default class Squares {
  private square: number;
  readonly squareOfSum: number;
  readonly sumOfSquares: number;
  readonly difference: number;
  constructor(square: number) {
    this.square = square;
    this.squareOfSum = Math.pow((this.square * (this.square + 1)) / 2, 2);
    this.sumOfSquares =
      (this.square * (this.square + 1) * (2 * this.square + 1)) / 6;
    this.difference = this.squareOfSum - this.sumOfSquares;
  }
}
